### PR TITLE
EOS-26011: Performance Stats: Expose a CSM REST API for exposing performance data with Prometheus support

### PR DESCRIPTION
**Pull request by Pranali04796**
Performance Stats: 
Signed-off-by: Pranali04796 <pranali.ugale@seagate.com>

# Problem Statement
Expose a CSM REST API for exposing performance data with Prometheus support
[EOS:26011](https://jts.seagate.com/browse/EOS-26011)
# Design
-  Expose REST api for metrics to be consumed by Prometheus
-  Currently metrics are fetched from static file.

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with Custom image- DONE 

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

image

### CODE REVIEW

**Review by rohitkolapkar — CHANGES_REQUESTED** [review]

File: csm/core/services/stats.py [hunk 1]
```diff
@@ -151,3 +153,9 @@ async def get_metrics(self, stats_id, metrics_list, from_t, to_t, interval,
         output["metrics"] = data_list
         Log.debug(f"Stats Request Output: {output}")
         return output
+
+    async def get_test_metrics(self):
+        file = open('/root/metrics.txt',mode='r')
```
[the remark this answers is not in the document]
Comment (Pranali04796) [reply csm/core/services/stats.py]: It is a static file. Added in repo.

File: csm/conf/etc/csm/csm.conf [hunk 2]
```diff
@@ -76,6 +76,7 @@ STATS:
         ssl_check: "false"
         interval: "10" # Flush interval in sec (Rate at which stats is stored)
         offset: "20" # offset in sec
+    telemetryAuth: "false"
```
Comment (rohitkolapkar) [inline csm/conf/etc/csm/csm.conf:79]: If you are not using this key - > 'telemetryAuth'.. then please remove it.

File: csm/core/blogic/const.py [hunk 3]
```diff
@@ -914,4 +914,5 @@
 AGENT_ENDPOINTS = 'CSM_SERVICE>CSM_AGENT>endpoints'
 AGENT_HOST = 'CSM_SERVICE>CSM_AGENT>host'
 AGENT_PORT = 'CSM_SERVICE>CSM_AGENT>port'
-AGENT_BASE_URL = 'CSM_SERVICE>CSM_AGENT>base_url'
\ No newline at end of file
+AGENT_BASE_URL = 'CSM_SERVICE>CSM_AGENT>base_url'
+TELEMENTRY_AUTH = 'STATS>telemetryAuth'
```
Comment (rohitkolapkar) [inline csm/core/blogic/const.py:918]: If you are not using TELEMENTRY_AUTH.. then please remove it.

**Review by rohitkolapkar — CHANGES_REQUESTED** [review]

File: csm/conf/etc/csm/csm.conf (hunk 2 above)
Comment (Pranali04796) [reply csm/conf/etc/csm/csm.conf]: Current code changes will not require it. Removed them.

File: csm/core/blogic/const.py (hunk 3 above)
Comment (Pranali04796) [reply csm/core/blogic/const.py]: removed

**Review by rohitkolapkar — APPROVED** [review]

**Review by kupranay — APPROVED** [review]

**Pull request merged**